usestdlibvars - http-status-code configuration not respected

### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc.).
- [X] Yes, I've tried with the standalone linter if available (e.g., gocritic, go vet, etc.). (https://golangci-lint.run/usage/linters/)

### Description of the problem

With the `http-status-code` configured as `false` for `usestdlibvars`, it should not report errors for use of numeric literals such as `200`. But with the latest version of `golangci-lint`, the error is still being reported. If I run the linter directly, it behaves as expected:

```console
$ usestdlibvars -http-status-code=false ./...
<succeeds without any output>

$ usestdlibvars ./...
/.../tusestdlibvars/main.go:15:24: "200" can be replaced by http.StatusOK
```

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 1.52.2 built with go1.20.2 from da04413 on 2023-03-23T16:18:48Z
```

</details>

### Configuration file

<details>

```console
$ cat .golangci.toml
[linters]
  disable-all = true
  enable = ["usestdlibvars"]

[linters-settings]
  [linters-settings.usestdlibvars]
    # Suggest the use of http.StatusXX.
    # covered by stylecheck ST1013.
    # Default: true
    http-status-code = false

```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.20.3 darwin/arm64
GO111MODULE=""
GOARCH="arm64"
GOBIN=""
GOCACHE="/Users/skaranth/Library/Caches/go-build"
GOENV="/Users/skaranth/Library/Application Support/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="arm64"
GOHOSTOS="darwin"
GOINSECURE=""
GOMODCACHE="/Users/skaranth/dev/work/go/pkg/mod"
GONOPROXY="_REDACTED_"
GONOSUMDB="_REDACTED_"
GOOS="darwin"
GOPATH="/Users/skaranth/dev/work/go"
GOPRIVATE="_REDACTED_"
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/opt/homebrew/Cellar/go/1.20.3/libexec"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/opt/homebrew/Cellar/go/1.20.3/libexec/pkg/tool/darwin_arm64"
GOVCS=""
GOVERSION="go1.20.3"
GCCGO="gccgo"
AR="ar"
CC="cc"
CXX="c++"
CGO_ENABLED="1"
GOMOD="/.../tusestdlibvars/go.mod"
GOWORK=""
CGO_CFLAGS="-O2 -g"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-O2 -g"
CGO_FFLAGS="-O2 -g"
CGO_LDFLAGS="-O2 -g"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -arch arm64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/zp/01m9f4ld29l4crxjfhhpnxxw0000gn/T/go-build112225028=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
INFO [config_reader] Config search paths: [./ /.../tusestdlibvars _REDACTED_ /Users /]
INFO [config_reader] Used config file .golangci.toml
INFO [lintersdb] Active 1 linters: [usestdlibvars]
INFO [loader] Go packages loading at mode 7 (compiled_files|files|name) took 145.543792ms
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 1.47575ms
INFO [linters_context/goanalysis] analyzers took 50.583µs with top 10 stages: inspect: 43.125µs, usestdlibvars: 7.458µs
INFO [runner] Processors filtering stat (out/in): max_same_issues: 1/1, path_shortener: 1/1, severity-rules: 1/1, cgo: 1/1, skip_files: 1/1, identifier_marker: 1/1, filename_unadjuster: 1/1, autogenerated_exclude: 1/1, sort_results: 1/1, max_from_linter: 1/1, source_code: 1/1, path_prettifier: 1/1, skip_dirs: 1/1, uniq_by_line: 1/1, diff: 1/1, max_per_file_from_linter: 1/1, fixer: 1/1, path_prefixer: 1/1, exclude: 1/1, exclude-rules: 1/1, nolint: 1/1
INFO [runner] processing took 112.709µs with stages: nolint: 30.75µs, path_prettifier: 21.375µs, autogenerated_exclude: 15.833µs, identifier_marker: 14.542µs, exclude-rules: 12.291µs, source_code: 10.459µs, skip_dirs: 3.167µs, max_same_issues: 834ns, cgo: 626ns, uniq_by_line: 583ns, path_shortener: 375ns, max_from_linter: 292ns, filename_unadjuster: 291ns, severity-rules: 250ns, max_per_file_from_linter: 249ns, skip_files: 209ns, diff: 208ns, exclude: 166ns, sort_results: 84ns, fixer: 83ns, path_prefixer: 42ns
INFO [runner] linters took 14.903583ms with stages: usestdlibvars: 14.753333ms
main.go:15:24: "200" can be replaced by http.StatusOK (usestdlibvars)
	if resp.StatusCode != 200 {
	                      ^
INFO File cache stats: 1 entries of total size 269B
INFO Memory: 3 samples, avg is 27.2MB, max is 27.4MB
INFO Execution took 168.664208ms
```

</details>

### Code example or link to a public repository

<details>

```go
package main

import (
	"fmt"
	"net/http"
)

func main() {
	resp, err := http.Get("https://gobyexample.com")
	if err != nil {
		panic(err)
	}
	defer resp.Body.Close()

	if resp.StatusCode != 200 {
		panic(resp.Status)
	}
	fmt.Println("Response status:", resp.Status)
}

```

</details>

--- a/pkg/config/linters_settings.go
+++ b/pkg/config/linters_settings.go
@@ -122,6 +122,10 @@ var defaultLintersSettings = LintersSettings{
 	Unparam: UnparamSettings{
 		Algo: "cha",
 	},
+	UseStdlibVars: UseStdlibVarsSettings{
+		HTTPMethod:     true,
+		HTTPStatusCode: true,
+	},
 	Varnamelen: VarnamelenSettings{
 		MaxDistance:   5,
 		MinNameLength: 3,

--- a/pkg/lint/lintersdb/manager.go
+++ b/pkg/lint/lintersdb/manager.go
@@ -252,6 +252,7 @@ func (m Manager) GetAllSupportedLinterConfigs() []*linter.Config {
 		thelperCfg = &m.cfg.LintersSettings.Thelper
 		unparamCfg = &m.cfg.LintersSettings.Unparam
 		unusedCfg = new(config.StaticCheckSettings)
+		usestdlibvars = &m.cfg.LintersSettings.UseStdlibVars
 		varcheckCfg = &m.cfg.LintersSettings.Varcheck
 		varnamelenCfg = &m.cfg.LintersSettings.Varnamelen
 		whitespaceCfg = &m.cfg.LintersSettings.Whitespace

--- a/test/testdata/configs/usestdlibvars_non_default.yml
+++ b/test/testdata/configs/usestdlibvars_non_default.yml
@@ -1,0 +1,12 @@
+linters-settings:
+  usestdlibvars:
+    http-method: false
+    http-status-code: false
+    time-weekday: true
+    time-month: true
+    time-layout: true
+    crypto-hash: true
+    default-rpc-path: true
+    sql-isolation-level: true
+    tls-signature-scheme: true
+    constant-kind: true

--- a/test/testdata/usestdlibvars_non_default.go
+++ b/test/testdata/usestdlibvars_non_default.go
@@ -1,0 +1,54 @@
+//golangcitest:args -Eusestdlibvars
+//golangcitest:config_path testdata/configs/usestdlibvars_non_default.yml
+package testdata
+
+import "net/http"
+
+func _200() {
+	_ = 200
+}
+
+func _200_1() {
+	var w http.ResponseWriter
+	w.WriteHeader(200)
+}
+
+const (
+	_ = "Bool"    // want `"Bool" can be replaced by constant\.Bool\.String\(\)`
+	_ = "Complex" // want `"Complex" can be replaced by constant\.Complex\.String\(\)`
+)
+
+const (
+	_ = "BLAKE2b-256" // want `"BLAKE2b-256" can be replaced by crypto\.BLAKE2b_256\.String\(\)`
+	_ = "BLAKE2b-384" // want `"BLAKE2b-384" can be replaced by crypto\.BLAKE2b_384\.String\(\)`
+)
+
+const (
+	_ = "/_goRPC_"   // want `"/_goRPC_" can be replaced by rpc\.DefaultRPCPath`
+	_ = "/debug/rpc" // want `"/debug/rpc" can be replaced by rpc\.DefaultDebugPath`
+)
+
+const (
+	_ = "Read Committed"   // want `"Read Committed" can be replaced by sql\.LevelReadCommitted\.String\(\)`
+	_ = "Read Uncommitted" // want `"Read Uncommitted" can be replaced by sql\.LevelReadUncommitted\.String\(\)`
+)
+
+const (
+	_ = "01/02 03:04:05PM '06 -0700" // want `"01/02 03:04:05PM '06 -0700" can be replaced by time\.Layout`
+	_ = "02 Jan 06 15:04 -0700"      // want `"02 Jan 06 15:04 -0700" can be replaced by time\.RFC822Z`
+)
+
+const (
+	_ = "April"  // want `"April" can be replaced by time\.April\.String\(\)`
+	_ = "August" // want `"August" can be replaced by time\.August\.String\(\)`
+)
+
+const (
+	_ = "Friday" // want `"Friday" can be replaced by time\.Friday\.String\(\)`
+	_ = "Monday" // want `"Monday" can be replaced by time\.Monday\.String\(\)`
+)
+
+const (
+	_ = "ECDSAWithP256AndSHA256" // want `"ECDSAWithP256AndSHA256" can be replaced by tls\.ECDSAWithP256AndSHA256\.String\(\)`
+	_ = "ECDSAWithP384AndSHA384" // want `"ECDSAWithP384AndSHA384" can be replaced by tls\.ECDSAWithP384AndSHA384\.String\(\)`
+)

